src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -274,25 +299,29 @@ private void processStart(DetailAST ast) {
         final int before = ast.getColumnNo() - 1;
         final int after = ast.getColumnNo() + 1;
 
-        // Need to handle two cases as in:
+        // Checks if generic needs to be preceded by a whitespace or not.
+        // Handles 3 cases as in:
         //
         //   public static <T> Callable<T> callable(Runnable task, T result)
         //                 ^           ^
-        //      ws reqd ---+           +--- whitespace NOT required
+        //   1. ws reqd ---+        2. +--- whitespace NOT required
         //
+        //   new <String>Object()
+        //       ^
+        //    3. +--- ws required
```

Comment:
Added this example in the comments as earlier `isGenericBeforeCtor()` (now `isGenericAfterNew()`) was added to be used in this method for identifying the cases where a preceding whitespace was required, but comment was not updated to showcase this new detection.